src/graphapi/opgraph.cpp

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/graphapi/opgraph.hpp>
+
+#include <unordered_map>
+
+namespace miopen {
+namespace graphapi {
+
+OpNode::~OpNode() = default;
+
+OpGraph OpGraphBuilder::build() &&
```

Comment:
Maybe I am missing something, but -- can you please explain a little why we want to allow this method for rvalues only?

Reply:
It's required to make `build` a bit more explicit about its state invalidation nature.
You can't call `build` twice.

Reply:
@atamazov : the idea is that we use `builder` as a temporary object to construct another object and then throw it away. I am not a fan but sometimes it's necessary when you cannot create the target object in one shot. Imagine for examples if graph nodes were being read from the file or network, then builder can capture them and build the graph when `build()` is called. 

In this case, it's partly due to how the Backend C API is, where each attribute is initialized one at a time.